Zend/zend_alloc.c

```diff
@@ -2632,6 +2632,15 @@ ZEND_API void* ZEND_FASTCALL _ecalloc(size_t nmemb, size_t size ZEND_FILE_LINE_D
 	return p;
 }
 
+ZEND_API void* ZEND_FASTCALL _ereallocarray(void* ptr, size_t nmemb, size_t size ZEND_FILE_LINE_DC ZEND_FILE_LINE_ORIG_DC)
+{
+	void *p;
+
+	size = zend_safe_address_guarded(nmemb, size, 0);
+	p = _erealloc(ptr, size ZEND_FILE_LINE_RELAY_CC ZEND_FILE_LINE_ORIG_RELAY_CC);
```

Comment:
This is essentially exactly the same as `safe_erealloc(ptr, nmemb, size, 0)`, no?

Reply:
good point :)

Reply:
But wasn't the point of `reallocarray` that in case of an overflow for `m * n` we don't touch the memory? I'm not completely sure what `erealloc` does with size `0`.

Reply:
I just noticed, `zend_safe_address_guarded` triggers a `zend_error_noreturn` on overflow, so checking for `0` doesn't make sense. I wonder if this PR makes sense at all, if we can essentially just do `safe_erealloc(ptr, nmemb, size, 0)`.

Reply:
yes you re right

Reply:
You could still switch the few cases form erealloc to erealloc_safe 🙂

Reply:
sure I prefer a new fresh PR ;-)